backend/alert/views.py

```diff
@@ -486,14 +517,23 @@ def get_queryset(self):
     def get_queryset_current(self):
         """
         Returns a superset of all active registrations (also includes cancelled registrations
-        from the current semester at the head of their resubscribe chains).
+        from the current semester at the head of their resubscribe chains). Returns at most 1
+        registration per section (if multiple candidate registrations for a certain section exist,
+        the registration with the later created_at value is chosen).
         """
-        return Registration.objects.filter(
+        registrations = Registration.objects.filter(
             user=self.request.user,
             deleted=False,
             resubscribed_to__isnull=True,
             section__course__semester=get_current_semester(),
         )
+        # Now resolve conflicts where multiple registrations exist for the same section
+        # (by taking the registration with the later created_at date)
+        return registrations.filter(
```

Comment:
Is this query efficient? Worried about the subquery here, if it's being executed all in the DB or if we're getting a list back and then sending it back to the DB. Could you inspect with the debug toolbar?

Reply:
Yeah looks like it's all one query (since the queryset is evaluated lazily)

```
SELECT "alert_registration"."id",
       "alert_registration"."created_at",
       "alert_registration"."original_created_at",
       "alert_registration"."updated_at",
       "alert_registration"."source",
       "alert_registration"."api_key_id",
       "alert_registration"."user_id",
       "alert_registration"."email",
       "alert_registration"."phone",
       "alert_registration"."section_id",
       "alert_registration"."cancelled",
       "alert_registration"."cancelled_at",
       "alert_registration"."deleted",
       "alert_registration"."deleted_at",
       "alert_registration"."auto_resubscribe",
       "alert_registration"."notification_sent",
       "alert_registration"."notification_sent_at",
       "alert_registration"."close_notification",
       "alert_registration"."close_notification_sent",
       "alert_registration"."close_notification_sent_at",
       "alert_registration"."notification_sent_by",
       "alert_registration"."close_notification_sent_by",
       "alert_registration"."resubscribed_from_id"
  FROM "alert_registration"
  LEFT OUTER JOIN "alert_registration" T2
    ON ("alert_registration"."id" = T2."resubscribed_from_id")
 INNER JOIN "courses_section"
    ON ("alert_registration"."section_id" = "courses_section"."id")
 INNER JOIN "courses_course"
    ON ("courses_section"."course_id" = "courses_course"."id")
 WHERE (NOT "alert_registration"."deleted" AND T2."id" IS NULL AND "courses_course"."semester" = '''2020C''' AND "alert_registration"."user_id" = '1' AND "alert_registration"."created_at" IN (SELECT MAX(U0."created_at") AS "max_created_at" FROM "alert_registration" U0 LEFT OUTER JOIN "alert_registration" U1 ON (U0."id" = U1."resubscribed_from_id") INNER JOIN "courses_section" U2 ON (U0."section_id" = U2."id") INNER JOIN "courses_course" U3 ON (U2."course_id" = U3."id") WHERE (NOT U0."deleted" AND U1."id" IS NULL AND U3."semester" = '''2020C''' AND U0."user_id" = '1') GROUP BY U0."section_id"))
```